nodepool/config/config_test.go

```diff
@@ -467,6 +467,30 @@ vpcCIDR: "10.1.0.0/16"
 `,
 		},
 		{
+			context: "WithSpotFleetWithExperimentalAwsEnvironment",
+			configYaml: minimalValidConfigYaml + `
+worker:
+  spotFleet:
+    targetCapacity: 10
+experimental:
+  awsEnvironment:
+    enabled: true
+`,
+			expectedErrorMessage: "The experimental feature `awsEnvironment` assumes a node pool is managed by an ASG rather than a Spot Fleet.",
+		},
+		{
+			context: "WithSpotFleetWithExperimentalWaitSignal",
+			configYaml: minimalValidConfigYaml + `
+worker:
+  spotFleet:
+    targetCapacity: 10
+experimental:
+  waitSignal:
+    enabled: true
+`,
+			expectedErrorMessage: "The experimental feature `waitSignal` assumes a node pool is managed by an ASG rather than a Spot Fleet.",
+		},
+		{
```

Comment:
FYI, this is how I tend to add minimum set of tests for error cases for each new feature.

* All the entries in the `parseErrorCases` slice are sorted in ascending order of `context`.
* `configYaml` is a complete `cluster.yaml` which is expected to fail loading